internal/api/handlers/operations_handler.go

```diff
@@ -147,6 +144,68 @@ func (h *OperationsHandler) GetOperation(ctx context.Context, request *api.GetOp
 	return &api.GetOperationResponse{Operation: convertedOp}, nil
 }
 
+func (h *OperationsHandler) parseListOperationsResponse(ctx context.Context, operationEntities []*operation.Operation) ([]*api.ListOperationItem, error) {
+	operationResponse, err := requestadapters.FromOperationsToListOperationsResponses(operationEntities)
+	if err != nil {
+		h.logger.Error("error parsing operations", zap.Error(err))
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	return operationResponse, nil
+}
+
+func (h *OperationsHandler) queryPendingOperations(ctx context.Context, schedulerName, sortingOrder string) ([]*operation.Operation, error) {
+	pendingOperationEntities, err := h.operationManager.ListSchedulerPendingOperations(ctx, schedulerName)
+	if err != nil {
+		h.logger.Error("error listing pending operations", zap.Error(err))
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	sortOperationsByCreatedAt(pendingOperationEntities, sortingOrder)
+
+	return pendingOperationEntities, nil
+}
+
+func (h *OperationsHandler) queryActiveOperations(ctx context.Context, schedulerName, sortingOrder string) ([]*operation.Operation, error) {
+	activeOperationEntities, err := h.operationManager.ListSchedulerActiveOperations(ctx, schedulerName)
+	if err != nil {
+		h.logger.Error("error listing active operations", zap.Error(err))
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	sortOperationsByCreatedAt(activeOperationEntities, sortingOrder)
+
+	return activeOperationEntities, nil
+}
+
+func (h *OperationsHandler) queryFinishedOperations(ctx context.Context, schedulerName, sortingOrder string, page, pageSize int64) ([]*operation.Operation, uint32, error) {
+	finishedOperationEntities, total, err := h.operationManager.ListSchedulerFinishedOperations(ctx, schedulerName, page-1, pageSize)
+	if err != nil {
+		h.logger.Error("error listing finished operations", zap.Error(err))
+		return nil, uint32(0), status.Error(codes.Unknown, err.Error())
+	}
+	sortOperationsByCreatedAt(finishedOperationEntities, sortingOrder)
+
+	return finishedOperationEntities, uint32(total), nil
+}
+
+func extractPaginationParameters(request *api.ListOperationsRequest) (uint32, uint32, error) {
+	page := request.GetPage()
+	pageSize := request.GetPerPage()
+	operationStage := request.Stage
+
+	if page == 0 {
+		page = 1
+	}
```

Comment:
Hello @mflilian and @guilhermocc here I think that is more appropriate to execute math decreasing `1` from the value and after validating if the number is valid (greater or equal to 1).